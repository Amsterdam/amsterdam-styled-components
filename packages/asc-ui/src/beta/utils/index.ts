import deprecatedWarning from './deprecatedWarning'

export {
  themeColor,
  srOnlyStyle,
  breakpoint,
  svgFill,
  fluidTypoStyle,
  typographyStyle,
  showAboveBackDrop,
  getValueFromTheme,
  themeSpacing,
} from './themeUtils'
export { valueFromObject } from './valueFromObject'
export { fromProps } from './fromProps'

export { deprecatedWarning }
