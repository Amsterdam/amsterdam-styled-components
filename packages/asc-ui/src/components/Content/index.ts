export { default as ContentStyle } from './ContentStyle'
export { default as Content } from './Content'
