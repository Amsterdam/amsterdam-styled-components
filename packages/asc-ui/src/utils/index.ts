export { default as fromTheme, valueFromObject } from './fromTheme'
export {
  color,
  getColorFromTheme,
  getTypographyFromTheme,
  focusStyleOutline,
  srOnlyStyle,
  breakpoint,
  svgFill,
  showHide,
  ShowHideTypes,
} from './themeUtils'
export { customCss, CustomCssPropsType, CustomCssPropsInterface } from './utils'
