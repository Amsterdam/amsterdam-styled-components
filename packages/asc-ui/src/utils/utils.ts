import { AbstractNode, IconDefinition } from '@datapunt/asc-icons/lib/types'
import * as React from 'react'

export function log(message: string) {
  if (!(process && process.env && process.env.NODE_ENV === 'production')) {
    console.error(`[icons-react]: ${message}.`)
  }
}

export function isIconDefinition(target: any): target is IconDefinition {
  return (
    typeof target === 'object' &&
    typeof target.name === 'string' &&
    typeof target.icon === 'object'
  )
}

export function normalizeAttrs(attrs: Attrs = {}): Attrs {
  return Object.keys(attrs).reduce((acc: Attrs, key) => {
    const val = attrs[key]
    switch (key) {
      case 'class':
        acc.className = val
        delete acc.class
        break
      default:
        acc[key] = val
    }
    return acc
  }, {})
}

export interface Attrs {
  [key: string]: string
}

export class MiniMap<V> {
  get size() {
    return Object.keys(this.collection).length
  }

  private collection: { [key: string]: V } = {}

  clear(): void {
    this.collection = {}
  }

  delete(key: string): boolean {
    return delete this.collection[key]
  }

  get(key: string): V | undefined {
    return this.collection[key]
  }

  has(key: string): boolean {
    return Boolean(this.collection[key])
  }

  set(key: string, value: V): this {
    this.collection[key] = value
    return this
  }
}

export function generate(
  node: AbstractNode,
  key: string,
  rootProps?: { [key: string]: any } | false,
): any {
  if (!rootProps) {
    return React.createElement(
      node.tag,
      { key, ...normalizeAttrs(node.attrs) },
      (node.children || []).map((child, index) =>
        generate(child, `${key}-${node.tag}-${index}`),
      ),
    )
  }
  return React.createElement(
    node.tag,
    {
      key,
      ...normalizeAttrs(node.attrs),
      ...rootProps,
    },
    (node.children || []).map((child, index) =>
      generate(child, `${key}-${node.tag}-${index}`),
    ),
  )
}
