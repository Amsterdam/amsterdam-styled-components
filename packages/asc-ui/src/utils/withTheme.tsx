import * as React from 'react'
import { shallow, render, mount } from 'enzyme'
import { ThemeProvider } from '../index'

export const renderWithTheme = (component: JSX.Element) =>
  render(<ThemeProvider>{component}</ThemeProvider>)

export const shallowWithTheme = (component: JSX.Element) =>
  shallow(<ThemeProvider>{component}</ThemeProvider>)

export const mountWithTheme = (component: JSX.Element) =>
  mount(<ThemeProvider>{component}</ThemeProvider>)
