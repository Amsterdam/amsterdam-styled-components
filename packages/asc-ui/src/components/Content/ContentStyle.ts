import styled from '@datapunt/asc-core'
import { customCss, CustomCssPropsInterface } from '../../utils'

export interface Props extends CustomCssPropsInterface {
  margin?: string
  padding?: string
}

export default styled.div<Props>`
  margin: ${({ margin }) => margin || '0'};
  padding: ${({ padding }) => padding || '0'};
  ${customCss}
`
