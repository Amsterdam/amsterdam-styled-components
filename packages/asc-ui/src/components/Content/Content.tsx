import React from 'react'
import ContentStyle, { Props as ContentStyleProps } from './ContentStyle'

export interface Props extends ContentStyleProps {
  StyledComponent?: any
}

const Content: React.FC<Props> = ({
  children,
  StyledComponent,
  ...otherProps
}) => <StyledComponent {...otherProps}>{children}</StyledComponent>

Content.defaultProps = {
  StyledComponent: ContentStyle,
}

export default Content
