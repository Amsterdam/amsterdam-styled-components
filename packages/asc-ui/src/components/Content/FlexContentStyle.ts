import styled from '@datapunt/asc-core'
import ContentStyle, { Props as ContentStyleProps } from './ContentStyle'
import { customCss } from '../../utils'

export interface Props extends ContentStyleProps {
  wrapReverse?: string
  noWrap?: string
  justifyContent?: string
  justifyCenter?: string
  alignItems?: string
  alignCenter?: string
  alignContent?: string
  column?: string
}

export default styled(ContentStyle)<Props>`
  display: flex;
  flex-wrap: ${props => {
    if (props.wrapReverse) return 'wrap-reverse'
    if (props.noWrap) return 'nowrap'
    return 'wrap'
  }};
  justify-content: ${props => {
    if (props.justifyContent) return props.justifyContent
    if (props.justifyCenter) return 'center'
    return 'flex-start'
  }};
  align-items: ${props => {
    if (props.alignItems) return props.alignItems
    if (props.alignCenter) return 'center'
    return 'flex-start'
  }};
  align-content: ${props => {
    if (props.alignContent) return props.alignContent
    return 'stretch'
  }};
  flex-direction: ${props => (props.column ? 'column' : 'row')};
  ${customCss}
`
