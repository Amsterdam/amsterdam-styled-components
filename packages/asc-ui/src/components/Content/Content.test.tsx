import * as React from 'react'
import Content from './Content'
import { renderWithTheme } from '../../utils/withTheme'
import 'jest-styled-components'

jest.useFakeTimers()

describe('TopBar', () => {
  let component: Cheerio

  beforeEach(() => {
    component = renderWithTheme(<Content>tcontent</Content>)
  })

  it('should render', () => {
    expect(component).toMatchSnapshot()
  })
})
