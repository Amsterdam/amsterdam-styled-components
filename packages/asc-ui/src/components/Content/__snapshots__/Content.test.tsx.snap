// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar should render 1`] = `
.c0 {
  margin: 0;
  padding: 0;
}

<div
  class="c0"
>
  tcontent
</div>
`;
